ansible_collections/arista/avd/docs/semantic-versioning.md

```diff
@@ -0,0 +1,216 @@
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- The resulting CloudVision configuration will be maintained, unless in rare cases, when addressing a bug.
+  - Breaking changes will be communicated in the release notes.
+
+### Role: eos_config_deploy_eapi
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- All role outputs follow SemVer.
+
+### Role: eos_validate_state
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- The generated reports and other artifacts may change between minor releases.
+  - Breaking changes to YAML/JSON/CSV outputs will be communicated in the release notes.
+
+### Role: dhcp_provisioner
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- Structured configuration output may change during minor releases to accommodate changes in the ansible-cvp collection.
+- The resulting DHCP configuration will be maintained, unless in rare cases, when addressing a bug.
+  - Breaking changes will be communicated in the release notes.
+
+### Role: build_output_folders
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- All role ouptus follow SemVer.
+
+### Ansible Plugins
+
+- No Ansible modules, action plugins, filter plugins, test plugins, or vars plugins are covered by SemVer.
+  - Functionality may change in minor releases to accommodate the project's needs.
+  - Breaking changes will be communicated in the release notes.
+
+## PyAVD
+
+!!! note
+
+    - All documented function input variables follow SemVer. Any breaking changes will, at a minimum, be communicated with a deprecation notice for 6 months.
+
+!!! warning
+
+    - No internal/undocumented functions are covered by SemVer, changes may also not be documented in release notes.
+
+### Function: validate_inputs()
+
+Inputs:
+
+- Function input variables follow SemVer.
+  - Any breaking changes will, at a minimum, be communicated with a deprecation notice for 6 months.
+
+### Function: get_avd_facts()
+
+Inputs:
+
+- Function input variables follow SemVer.
+  - Any breaking changes will, at a minimum, be communicated with a deprecation notice for 6 months.
+
+Outputs:
+
+- `avd_facts` may change between minor releases.
```

Comment:
Can't this one also cause breaking changes?

Reply:
If people are using our facts (`switch.*`) then yes.
This is called out in the docs already:
```
While all templates can leverage the internal switch facts (switch.*) to customize the interface descriptions, the values are not part of the officially supported data models, and may change without notice.
```

Reply:
So it may cause breaking changes (but not to AVD as AVD would be adapting to the changes already). 
Perhaps a call out here as a courtesy?

Reply:
addresses with [5e959f0](https://github.com/aristanetworks/avd/pull/3417/commits/5e959f0099a8581aa36f926659d4c9e10b59747a)